naga/src/front/wgsl/error.rs

```diff
@@ -295,12 +295,29 @@ pub(crate) enum Error<'a> {
     DiagnosticInvalidSeverity {
         severity_control_name_span: Span,
     },
+    DiagnosticDuplicateTriggeringRule {
+        triggering_rule: FilterableTriggeringRule,
+        triggering_rule_spans: [Span; 2],
+    },
```

Comment:
Would it make sense to simply let this be:

```rust
DiagnosticDuplicateTriggeringRule(ConflictingDiagnosticRuleError),
```

rather than duplicating the definition of `ConflictingDiagnosticRuleError` as the enum variant?

There's probably a reason, but I'm surprised we're not using `thiserror` here; I expected this to be a `#[from]` variant.

Reply:
I raised the premise of your question in <https://github.com/gfx-rs/wgpu/pull/6148#discussion_r1823756480>. Not sure if you saw it, but 🤷🏻‍♂️ I'm fine moving this to simply be within `DiagnosticDuplicateTriggeringRule`.

Implemented as suggested with the force-pushed 4fc60fe9.